Content.Shared/Preferences/PlayerPreferences.cs

```diff
@@ -45,6 +51,11 @@ public ICharacterProfile GetProfile(int index)
 
         public int IndexOfCharacter(ICharacterProfile profile)
         {
+            if (profile == null)
+            {
+                return _characters.Count;
```

Comment:
Should probably
```suggestion
                return _characters.Count - 1;
```
because DeleteCharacter(null) will without access _characters[Count] which will ree

Reply:
Fixed by having Delete not use Update with a null profile.